pkg/controller/identity/iamrolepolicyattachment/controller.go

```diff
@@ -127,33 +127,31 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 
 	cr.SetConditions(runtimev1alpha1.Available())
 
-	cr.UpdateExternalStatus(*attachedPolicyObject)
+	iam.UpdateRolePolicyExternalStatus(cr, *attachedPolicyObject)
```

Comment:
Here we are already updating the status in the `Observe` method (contrary to the iam role controller), but it would be nice to have a `GenerateObservation` function that returns the `AtProvider` observation rather than passing in the object to have its `AtProvider` status set in the client.